### ISSUE TITLE
Pylance false positive with simple keywords unpacking

### ISSUE THREAD

**Issue by smojef** [microsoft/pyright#10698]
**Describe the bug**
I get this error : 
Argument of type "int" cannot be assigned to parameter "sarg" of type "str" in function "fn1"
with the following minimal code:

**Code or Screenshots**
```python
def fn0() -> None:
    kw = dict(iarg=1)
    fn1(**kw)  # Argument of type "int" cannot be assigned to parameter "sarg" of type "str" in function "fn1"
    
def fn1(sarg:str='string', **kw) -> None:
    print(locals())
```
**VS Code extension or command-line**
VS Code extension, ms-pyright.pyright 1.1.403

**Comment by erictraut** [comment]
This is the intended behavior, so I don't consider this a bug.

The type of `kw` is `dict[str, int]`. When this dictionary is unpacked, it supplies arguments for an unknown set of keyword parameters in the target function. One of those parameters is `sarg` which has a declared type of `str`. Since `int` isn't assignable to `str`, this is a type violation.

For comparison, mypy and other Python type checkers emit an error here as well.

There are a few ways you can fix this type violation in your code. Perhaps the easiest is to change the signature of `fn1` to indicate that the first argument is positional-only. That way, there's no way that the unpacked dictionary can supply an argument for the first parameter.

```python
def fn1(sarg: str = "string", /, **kw) -> None:
    ...
```

**Comment by erictraut** [comment]
I should also mention that there have been recent discussions about standardizing the behavior of argument unpacking for calls, but consensus has not yet been reached. If you are interested in contributing to that discussion in the Python typing forum, here is a [link](https://discuss.python.org/t/standardizing-argument-unpacking-behaviors-for-calls/80747).

**Comment by smojef** [comment]
Hi @erictraut, thanks for the prompt response. I focused on the minimal code to reproduce the error, but the real issue becomes clear with these practical examples:

**Example 1: Common kwargs pattern**
```python
def func_with_catch_all(file, title='Default Title', **cfg):
    print(locals())

cfg_kw = dict(line_size=2, marker_size=10)
func_with_catch_all('data.txt', **cfg_kw)
# Error: Argument of type "int" cannot be assigned to parameter "title"
```

**Example 2: Multiple dict unpacking**
```python
def func_with_many_args(x, y, marker_size=5, marker_color='#ffffff', line_size=1, line_color='#ffffff'):
    print(locals())

marker_kw = dict(marker_size=10, marker_color='#ff0000')
line_kw = dict(line_size=2, line_color='#0000ff')
func_with_many_args(1, 2, **line_kw, **marker_kw)
# Multiple errors about int|str not being assignable to specific types
```

These are extremely common patterns in Python (especially in libraries like matplotlib, pandas, etc.), and the current behavior makes pyright unusable for such code.

**Key issues:**
1. **Pyright has the static information** to know that `cfg_kw` contains `{'line_size': 2, 'marker_size': 10}` - it should be able to determine that `title` won't be affected
2. **The `**cfg` parameter exists specifically** to catch these arguments, yet pyright assumes they'll somehow interfere with typed parameters
3. **Multiple dict unpacking** becomes completely broken with union types (int|str) that make no sense

This isn't about edge cases - this breaks fundamental Python patterns that work perfectly at runtime. The type checker should be helping developers, not blocking idiomatic code. 

Would it be possible to reconsider this as a bug rather than intended behavior? Or at minimum, provide a way to disable this specific check (not all argument type checking, but only the unpacking scenarios)?

Most developers are comfortable with a type checker that doesn't catch some complex edge cases (false negatives), but false positives that require code changes or pragma comments for perfectly working code significantly hurt the developer experience.
